packages/btcindexer/src/btcindexer.ts

```diff
@@ -340,12 +367,12 @@ export class Indexer implements Storage {
 					!proof ||
 					(block.merkleRoot !== undefined && !block.merkleRoot.equals(calculatedRoot))
 				) {
-					console.warn(
-						`WARN: Failed to generate a valid merkle proof for TX ${txId}. Root mismatch.`,
-						`Block root: ${block.merkleRoot?.toString(
-							"hex",
-						)}, Calculated: ${calculatedRoot.toString("hex")}`,
-					);
+					console.warn({
+						msg: "Failed to generate a valid merkle proof. Root mismatch.",
```

Comment:
This is when a block is not finalized, correct? Could you add more info why this happens to the log?